internal/common/errors.go

```diff
@@ -51,11 +51,22 @@ func CustomErrorHandler(ctx *fiber.Ctx, err error) error {
 		}
 	}
 
-	err = ctx.Status(problemJSON.Status).JSON(problemJSON)
+	ctx.Status(problemJSON.Status)
 
-	// Needs to be after the call to JSON(), to override the
-	// automatic Content-Type
-	ctx.Set(fiber.HeaderContentType, "application/problem+json")
+	if problemJSON.ValidationErrors != nil {
+		err = ctx.JSON(fiber.Map{
+			"title":            problemJSON.Title,
+			"detail":           problemJSON.Detail,
+			"status":           problemJSON.Status,
+			"validationErrors": problemJSON.ValidationErrors,
+		}, "application/problem+json")
+	} else {
+		err = ctx.JSON(fiber.Map{
+			"title":  problemJSON.Title,
+			"detail": problemJSON.Detail,
+			"status": problemJSON.Status,
+		}, "application/problem+json")
+	}
 
 	return err
```

Comment:
I think we can simplify this and have JSON be serialized with the rules specified in `ProblemJSON`'s struct fields, as we already have those: https://github.com/italia/developers-italia-api/blob/main/internal/common/problem_json.go#L7-L13

Something like:

```suggestion
	return ctx.JSON(problemJSON, "application/problem+json")
```